### PR TITLE
Automatic port of System.Reflection.Metadata.Ecma335

### PR DESCRIPTION
**Pull request by carlossanlop**
Area owners @tmat @nguerrera can you please review this PR? These comments were automatically ported directly from triple slash.

### CODE REVIEW

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 1]
```diff
@@ -101,9 +101,9 @@
         <Parameter Name="namedArguments" Type="System.Action&lt;System.Reflection.Metadata.Ecma335.CustomAttributeNamedArgumentsEncoder&gt;" Index="1" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="fixedArguments">To be added.</param>
-        <param name="namedArguments">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="fixedArguments">Called first, to encode fixed arguments.</param>
+        <param name="namedArguments">Called second, to encode named arguments.</param>
+        <summary>Encodes Custom Attribute Signature blob.</summary>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: capitalization:

```suggestion
        <summary>Encodes custom attribute signature blob.</summary>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 2]
```diff
@@ -158,8 +159,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Encodes Field Signature blob.</summary>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: same:

```suggestion
        <summary>Encodes field signature blob.</summary>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 3]
```diff
@@ -186,9 +187,9 @@
         <Parameter Name="variableCount" Type="System.Int32" Index="0" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="variableCount">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="variableCount">Number of local variables.</param>
+        <summary>Encodes Local Variable Signature.</summary>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: same:

```suggestion
        <summary>Encodes local variable signature.</summary>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 4]
```diff
@@ -216,11 +217,12 @@
...
-        <param name="genericParameterCount">To be added.</param>
-        <param name="isInstanceMethod">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="convention">Calling convention.</param>
+        <param name="genericParameterCount">Number of generic parameters.</param>
+        <param name="isInstanceMethod">
+          <see langword="true" /> to encode an instance method signature, <see langword="false" /> to encode a static method signature.</param>
+        <summary>Encodes Method Signature blob.</summary>
+        <returns>An Encoder of the rest of the signature including return value and parameters.</returns>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: same:

```suggestion
        <summary>Encodes method signature blob.</summary>
        <returns>An encoder of the rest of the signature including return value and parameters.</returns>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 5]
```diff
@@ -247,9 +249,9 @@
         <Parameter Name="genericArgumentCount" Type="System.Int32" Index="0" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="genericArgumentCount">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="genericArgumentCount">Number of generic arguments.</param>
+        <summary>Encodes Method Specification Signature blob.</summary>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: same:

```suggestion
        <summary>Encodes method specification signature blob.</summary>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 6]
```diff
@@ -276,9 +278,9 @@
         <Parameter Name="argumentCount" Type="System.Int32" Index="0" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="argumentCount">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="argumentCount">Number of arguments in the set.</param>
+        <summary>Encodes Permission Set arguments.</summary>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: same:

```suggestion
        <summary>Encodes permission set arguments.</summary>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 7]
```diff
@@ -305,9 +307,9 @@
         <Parameter Name="attributeCount" Type="System.Int32" Index="0" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="attributeCount">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="attributeCount">Number of attributes in the set.</param>
+        <summary>Encodes a Permission Set blob.</summary>
+        <returns>Permission Set encoder.</returns>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: same:

```suggestion
        <summary>Encodes a permission set blob.</summary>
        <returns>Permission set encoder.</returns>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 8]
```diff
@@ -333,9 +335,10 @@
         <Parameter Name="isInstanceProperty" Type="System.Boolean" Index="0" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="isInstanceProperty">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="isInstanceProperty">
+          <see langword="true" /> to encode an instance property signature, <see langword="false" /> to encode a static property signature.</param>
+        <summary>Encodes Property Signature blob.</summary>
+        <returns>An Encoder of the rest of the signature including return value and parameters, which has the same structure as Method Signature.</returns>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: same:

```suggestion
        <summary>Encodes property signature blob.</summary>
        <returns>An Encoder of the rest of the signature including return value and parameters, which has the same structure as method signature.</returns>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 9]
```diff
@@ -360,8 +363,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Encodes Type Specification Signature.</summary>
+        <returns>Type encoder of the structured type represented by the Type Specification (it shall not encode a primitive type).</returns>
```
Comment (BillWagner) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: one more:

```suggestion
        <summary>Encodes type specification signature.</summary>
        <returns>Type encoder of the structured type represented by the type specification (it shall not encode a primitive type).</returns>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: Thanks. I feel like this also needs to disambiguate with the rest of the overloads. How about this?:

```suggestion
        <summary>Encodes the custom attribute signature blob using the specified argument encoding actions.</summary>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 10]
```diff
@@ -131,9 +131,10 @@
         <Parameter Name="namedArguments" Type="System.Reflection.Metadata.Ecma335.CustomAttributeNamedArgumentsEncoder" RefType="out" Index="1" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="fixedArguments">To be added.</param>
-        <param name="namedArguments">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="fixedArguments">Use first, to encode fixed arguments.</param>
+        <param name="namedArguments">Use second, to encode named arguments.</param>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: I think I prefer using the word called, like in the first overload above:
```suggestion
        <param name="fixedArguments">Called first, to encode fixed arguments.</param>
        <param name="namedArguments">Called second, to encode named arguments.</param>
```

File: xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml [hunk 11]
```diff
@@ -131,9 +131,10 @@
...
+        <summary>Encodes Custom Attribute Signature blob.
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System.Reflection.Metadata.Ecma335/BlobEncoder.xml]: Same as above, making sure we disambiguate:

**Comment by carlossanlop** [comment]
@tmat @nguerrera I feel like the wording in these comments can be improved. Can you please help reword whatever's necessary?

**Pull request merged**